eclass/wxwidgets.eclass

```diff
@@ -66,49 +81,31 @@ esac
 
 # @FUNCTION: setup-wxwidgets
 # @DESCRIPTION:
-# Call this in your ebuild to set up the environment for wxGTK.  Besides
-# controlling the wx-config wrapper this exports WX_CONFIG containing
-# the path to the config in case it needs to be passed to a build system.
+# Call this in your ebuild to set up the environment for wxGTK in src_configure.
+# Besides controlling the wx-config wrapper this exports WX_CONFIG containing
+# the path to the config in case it needs to be passed to the build system.
 #
-# In wxGTK-2.9 and later it also controls the level of debugging output
-# from the libraries.  In these versions debugging features are enabled
-# by default and need to be disabled at the package level.  Because this
-# causes many warning dialogs to pop up during runtime we add -DNDEBUG to
-# CPPFLAGS to disable debugging features (unless your ebuild has a debug
-# USE flag and it's enabled).  If you don't like this behavior you can set
-# WX_DISABLE_NDEBUG to override it.
+# This function also controls the level of debugging output from the libraries.
+# Debugging features are enabled by default and need to be disabled at the
+# package level. Because this causes many warning dialogs to pop up during
+# runtime we add -DNDEBUG to CPPFLAGS to disable debugging features (unless
+# your ebuild has a debug USE flag and it's enabled). If you don't like this
+# behavior you can set WX_DISABLE_NDEBUG to override it.
 #
-# See: http://docs.wxwidgets.org/trunk/overview_debugging.html
-
+# See: https://docs.wxwidgets.org/trunk/overview_debugging.html
 setup-wxwidgets() {
 	local w wxtoolkit wxdebug wxconf
 
-	[[ -z ${WX_GTK_VER} ]] \
-		&& die "WX_GTK_VER must be set before calling $FUNCNAME."
-
-	case "${WX_GTK_VER}" in
-		3.0-gtk3)
-			wxtoolkit=gtk3
-			if [[ -z ${WX_DISABLE_NDEBUG} ]]; then
-				( in_iuse debug && use debug ) || append-cppflags -DNDEBUG
-			fi
-			;;
-		2.9|3.0)
-			wxtoolkit=gtk2
-			if [[ -z ${WX_DISABLE_NDEBUG} ]]; then
-				( in_iuse debug && use debug ) || append-cppflags -DNDEBUG
-			fi
-			;;
-		2.8)
-			wxtoolkit=gtk2
-			wxdebug="release-"
-			has_version x11-libs/wxGTK:${WX_GTK_VER}[debug] && wxdebug="debug-"
-			;;
-		*)
-			die "Invalid WX_GTK_VER: must be set to a valid wxGTK SLOT"
-			;;
+	case ${WX_GTK_VER} in
+		3.0-gtk3) wxtoolkit=gtk3 ;;
+		3.0)      wxtoolkit=gtk2 ;;
```

Comment:
Consider adding an eqawarn nag on the gtk2 variants? We're already filing [bugs](https://bugs.gentoo.org/618642) but this is a useful nudge.